sig-architecture/charter.md

```diff
@@ -1,40 +1,175 @@
+    <td>Test list, Test guidelines</td>
+  </tr>
+  <tr>
+    <td>Architectural governance</td>
+    <td>Establishing and documenting design principles, documenting and evolving the system architecture, reviewing, curating, and documenting new extension patterns</td>
+    <td>Kubernetes Design and Architecture, Design principles</td>
+  </tr>
+  <tr>
+    <td>API governance</td>
+    <td>Establishing and documenting conventions for system and user-facing APIs, define and operate the APl review process, final API implementation consistency validation, co-own top-level API directories with API machinery</td>
+    <td>API conventions, API linter</td>
+  </tr>
+  <tr>
+    <td>Deprecation policy</td>
+    <td>Maintaining, evolving, and enforcing the deprecation policy</td>
+    <td>Deprecation policy</td>
+  </tr>
+  <tr>
+    <td>Code organization</td>
+    <td>Overall code organization, including github repositories and branching methodology, top-level and pkg OWNERS of kubernetes/kubernetes, kubernetes-template-project</td>
+    <td>kubernetes-template-project</td>
+  </tr>
+  <tr>
+    <td>KEP</td>
+    <td>Develop and drive technical enhancement review process</td>
+    <td>Process, Template</td>
+  </tr>
+</table>
+
+
+## Roles
+
+The following roles are required for the SIG to properly function. In the event that any role is unfilled, the SIG will make a best effort to fill it, and any decisions reliant on a missing role will be postponed until the role is filled.
+
+* **Chair**
+
+    * Number: 2-3
+
+    * Run operations and processes governing the SIG
+
+    * An initial set of chairs was established at the time the SIG was founded.
+
+    * Chairs *MAY* decide to step down at anytime and propose a replacement, who must be approved by all of the other chairs. This *SHOULD* be supported by a majority of SIG Members.
+
+    * Chairs *MAY* select additional chairs by consensus. This *SHOULD* be supported by a majority of SIG Members.
+
+    * Chairs *MUST* remain active in the role and are automatically removed from the position if they are unresponsive for > 3 months and *MAY* be removed by consensus of the other Chairs and all of the Technical leads if not proactively working with other Chairs to fulfill responsibilities.
+
+    * A majority of chairs cannot be from a single company. If a chair’s change of company affiliation causes this threshold to be exceeded, one of the chairs must step down and propose a replacement, as per the procedure above.
+
+    * Defined in[ sigs.yaml](https://github.com/kubernetes/community/blob/master/sigs.yaml#L1454)
+
+* SIG Technical Leads
+
+    * Number: 3
+
+    * Establish new subprojects, and retire existing subprojects
+
+    * Resolve cross-subproject technical issues and decisions, and escalations from subprojects
+
+    * Decision-making MUST be by consensus. It’s particularly important for the technical leads to provide cohesive technical guidance for the project as a whole.
```

Comment:
It's important to have a fallback plan when consensus cannot be reached within a reasonable time, to prevent granting implicit veto rights to any tech lead.  I propose 2/3 majority vote as that fallback.

Reply:
More comprehensive discussion here: https://docs.google.com/document/d/1p4quG5qzIllQ52eKIb7mGrgaiRgQeiu70iJi9MZ9C8c/edit?disco=AAAAB7Yujow

Here are the most pertinent points, repeated:

Quinton: I'd be fully in support of language seeking diversity in general, and company diversity in particular. A few other points to add (most of which I raised at the meeting):

1. The number stated above is three. My strong suggestion of avoiding single-company majority was made primarily in that context, although it works well for other numbers too (i.e. it conveys what I was explicitly intending to convey), e.g. 1 of 2 is not a majority, 2 of 3 is both a majority and a super-majority, 2 of 4 is not a majority, etc.
2. I feel quite strongly that we need to set an example amongst SIG's by not having a single company hold a majority (or super-majority) of Chair or Tech Lead seats. As currently stated, a super-majority would give a single company the ability select additional chairs, make majority decisions amongst chairs, and disproportionately drive conversations. I don't think we want any of these.
3. I realize that currently Jaice and Brian are the chairs of this SIG, and both work for Google, and therefore one of the two would need to be replaced by a non-Googler in order to achieve the stated goal of non-majority. This may be uncomfortable, but I think it's worth the discomfort in order to set a good example.

Jaice, could you explain your rationale for wanting to keep the wording as is?

Jaice: 
Quinton, thank you for your continued concern here. In short I believe any of the hegemony concerns can be allayed through requiring consensus on decisions. This would give any company veto power on decisions, regardless of who has the majority of membership. Also, I would argue that if the chair position is "powerful" then we have done this incorrectly. People should not be seeking the position to have control or bragging rights, and we need to disincentivize that in any way possible, or we are walking into a known anti-pattern. The chair is a servant leadership position that handles SIG-related administrivia. Even the agenda document has no write restrictions, so the only privileged access a chair has is to the Zoom account and lead-related mailing lists. I would welcome more voices and opinions here.

Quinton: 
s the wording stands right now, fallback in the case of lack of (lazy) consensus is a majority (or supermajority) vote by chairs. I think that's the right approach and preferable to vetos. I would just like to ensure that the majority or supermajority vote cannot be carried by a single company, against all others. That's why I want "a majority of chairs cannot be held by a single company". 

I still do not understand the argument against that. Sure we could cling to the idea of allowing a single company to hold the supermajority of chair positions, and adjust all the other rules to accommodate that (e.g. allowing vetos or requiring consensus). But that's far more fraught with problems, IMO.

Jaice:
Quinton, I gave this a lot of thought and I am willing to withdraw my objections based on our intent to do a 6 month refresh of this document. I understand the implication on the current structure will mean either Brian or myself stepping down. I want for us to work on making this position servant leadership, with no privileges attached. I concede that in the current implementation that is not as true as I wished it was. We can discuss this in greater detail at the next SIG meeting.